### DISCUSSION TITLE
Compilation errors

### DISCUSSION THREAD

**Discussion by win7463** [rtl-airband/RTLSDR-Airband#423] [Q&A]
I am getting the following when trying to compile. I am running on a PI4 and following the instructions on the wiki. Can anyone help me figure out what I am doing wrong?

/home/jwinship/RTLSDR-Airband-4.1.1/src/output.cpp: In function ‘void shout_setup(icecast_data*, mix_modes)’:
/home/jwinship/RTLSDR-Airband-4.1.1/src/output.cpp:81:29: error: ‘int shout_set_format(shout_t*, unsigned int)’ is deprecated: Use shout_set_content_format() [-Werror=deprecated-declarations]
   81 |         if (shout_set_format(shouttemp, SHOUT_FORMAT_MP3) != SHOUTERR_SUCCESS){
      |             ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/jwinship/RTLSDR-Airband-4.1.1/src/output.cpp:30:
/usr/include/shout/shout.h:361:5: note: declared here
  361 | int shout_set_format(shout_t *self, unsigned int format) SHOUT_ATTR_F_DEPRECATED("Use shout_set_content_format()"); // obsolete
      |     ^~~~~~~~~~~~~~~~
/home/jwinship/RTLSDR-Airband-4.1.1/src/output.cpp:84:43: error: ‘int shout_set_name(shout_t*, const char*)’ is deprecated: use shout_set_meta() with SHOUT_META_NAME [-Werror=deprecated-declarations]
   84 |         if(icecast->name && shout_set_name(shouttemp, icecast->name) != SHOUTERR_SUCCESS) {
      |                             ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/shout/shout.h:348:5: note: declared here
  348 | int shout_set_name(shout_t *self, const char *name) SHOUT_ATTR_F_DEPRECATED("use shout_set_meta() with SHOUT_META_NAME"); // obsolete
      |     ^~~~~~~~~~~~~~
/home/jwinship/RTLSDR-Airband-4.1.1/src/output.cpp:87:45: error: ‘int shout_set_genre(shout_t*, const char*)’ is deprecated: use shout_set_meta() with SHOUT_META_GENRE [-Werror=deprecated-declarations]
   87 |         if(icecast->genre && shout_set_genre(shouttemp, icecast->genre) != SHOUTERR_SUCCESS) {
      |                              ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/shout/shout.h:354:5: note: declared here
  354 | int shout_set_genre(shout_t *self, const char *genre) SHOUT_ATTR_F_DEPRECATED("use shout_set_meta() with SHOUT_META_GENRE"); // obsolete
      |     ^~~~~~~~~~~~~~~
/home/jwinship/RTLSDR-Airband-4.1.1/src/output.cpp:90:57: error: ‘int shout_set_description(shout_t*, const char*)’ is deprecated: use shout_set_meta() with SHOUT_META_DESCRIPTION [-Werror=deprecated-declarations]
   90 |         if(icecast->description && shout_set_description(shouttemp, icecast->description) != SHOUTERR_SUCCESS) {
      |                                    ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/shout/shout.h:357:5: note: declared here
  357 | int shout_set_description(shout_t *self, const char *description) SHOUT_ATTR_F_DEPRECATED("use shout_set_meta() with SHOUT_META_DESCRIPTION"); // obsolete
      |     ^~~~~~~~~~~~~~~~~~~~~
/home/jwinship/RTLSDR-Airband-4.1.1/src/output.cpp: In function ‘void process_outputs(channel_t*, int)’:
/home/jwinship/RTLSDR-Airband-4.1.1/src/output.cpp:479:55: error: ‘int shout_set_metadata(shout_t*, shout_metadata_t*)’ is deprecated: Use shout_set_metadata_utf8() [-Werror=deprecated-declarations]
  479 |                                 if (shout_set_metadata(icecast->shout, meta) != SHOUTERR_SUCCESS) {
      |                                     ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
/usr/include/shout/shout.h:367:5: note: declared here
  367 | int shout_set_metadata(shout_t *self, shout_metadata_t *metadata) SHOUT_ATTR_F_WARN_UNUSED_RESULT SHOUT_ATTR_F_DEPRECATED("Use shout_set_metadata_utf8()");
      |     ^~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [src/CMakeFiles/rtl_airband_base.dir/build.make:153: src/CMakeFiles/rtl_airband_base.dir/output.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:100: src/CMakeFiles/rtl_airband_base.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
jwinship@raspberrypi:~/RTLSDR-Airband-4.1.1/build $ sudo make install
[  5%] Generating version.cpp, _version.cpp
[ 11%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/output.cpp.o
/home/jwinship/RTLSDR-Airband-4.1.1/src/output.cpp: In function ‘void shout_setup(icecast_data*, mix_modes)’:
/home/jwinship/RTLSDR-Airband-4.1.1/src/output.cpp:81:29: error: ‘int shout_set_format(shout_t*, unsigned int)’ is deprecated: Use shout_set_content_format() [-Werror=deprecated-declarations]
   81 |         if (shout_set_format(shouttemp, SHOUT_FORMAT_MP3) != SHOUTERR_SUCCESS){
      |             ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/jwinship/RTLSDR-Airband-4.1.1/src/output.cpp:30:
/usr/include/shout/shout.h:361:5: note: declared here
  361 | int shout_set_format(shout_t *self, unsigned int format) SHOUT_ATTR_F_DEPRECATED("Use shout_set_content_format()"); // obsolete
      |     ^~~~~~~~~~~~~~~~
/home/jwinship/RTLSDR-Airband-4.1.1/src/output.cpp:84:43: error: ‘int shout_set_name(shout_t*, const char*)’ is deprecated: use shout_set_meta() with SHOUT_META_NAME [-Werror=deprecated-declarations]
   84 |         if(icecast->name && shout_set_name(shouttemp, icecast->name) != SHOUTERR_SUCCESS) {
      |                             ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/shout/shout.h:348:5: note: declared here
  348 | int shout_set_name(shout_t *self, const char *name) SHOUT_ATTR_F_DEPRECATED("use shout_set_meta() with SHOUT_META_NAME"); // obsolete
      |     ^~~~~~~~~~~~~~
/home/jwinship/RTLSDR-Airband-4.1.1/src/output.cpp:87:45: error: ‘int shout_set_genre(shout_t*, const char*)’ is deprecated: use shout_set_meta() with SHOUT_META_GENRE [-Werror=deprecated-declarations]
   87 |         if(icecast->genre && shout_set_genre(shouttemp, icecast->genre) != SHOUTERR_SUCCESS) {
      |                              ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/shout/shout.h:354:5: note: declared here
  354 | int shout_set_genre(shout_t *self, const char *genre) SHOUT_ATTR_F_DEPRECATED("use shout_set_meta() with SHOUT_META_GENRE"); // obsolete
      |     ^~~~~~~~~~~~~~~
/home/jwinship/RTLSDR-Airband-4.1.1/src/output.cpp:90:57: error: ‘int shout_set_description(shout_t*, const char*)’ is deprecated: use shout_set_meta() with SHOUT_META_DESCRIPTION [-Werror=deprecated-declarations]
   90 |         if(icecast->description && shout_set_description(shouttemp, icecast->description) != SHOUTERR_SUCCESS) {
      |                                    ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/shout/shout.h:357:5: note: declared here
  357 | int shout_set_description(shout_t *self, const char *description) SHOUT_ATTR_F_DEPRECATED("use shout_set_meta() with SHOUT_META_DESCRIPTION"); // obsolete
      |     ^~~~~~~~~~~~~~~~~~~~~
/home/jwinship/RTLSDR-Airband-4.1.1/src/output.cpp: In function ‘void process_outputs(channel_t*, int)’:
/home/jwinship/RTLSDR-Airband-4.1.1/src/output.cpp:479:55: error: ‘int shout_set_metadata(shout_t*, shout_metadata_t*)’ is deprecated: Use shout_set_metadata_utf8() [-Werror=deprecated-declarations]
  479 |                                 if (shout_set_metadata(icecast->shout, meta) != SHOUTERR_SUCCESS) {
      |                                     ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
/usr/include/shout/shout.h:367:5: note: declared here
  367 | int shout_set_metadata(shout_t *self, shout_metadata_t *metadata) SHOUT_ATTR_F_WARN_UNUSED_RESULT SHOUT_ATTR_F_DEPRECATED("Use shout_set_metadata_utf8()");
      |     ^~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [src/CMakeFiles/rtl_airband_base.dir/build.make:153: src/CMakeFiles/rtl_airband_base.dir/output.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:100: src/CMakeFiles/rtl_airband_base.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

**Comment by charlie-foxtrot** [comment]
can you do a fresh `cmake` and post the output?  that should show what version of `libshout` it is finding.

after that can you switch to the `unstable` branch?

this looks like #381 that was fixed in #382 but then has additional changes in the (not yet merged) #422

**Reply by win7463** [reply]
I did a install for libshout3.dev and this is what I get. Will do the fresh cmake now as requested.

sudo apt-get install libshout3.dev
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'libshout3-dev' for regex 'libshout3.dev'
libshout3-dev is already the newest version (2.4.6-1+b1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.

**Reply by win7463** [reply]
win7463@raspberrypi:~/RTLSDR-Airband-4.1.1/build $ cmake ../ -DPLATFORM=armv8-generic
-- The CXX compiler identification is GNU 12.2.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Build type not specified: defaulting to Release
-- Performing Test CXX_HAS_PTHREAD
-- Performing Test CXX_HAS_PTHREAD - Success
-- Performing Test CXX_HAS_FFAST_MATH
-- Performing Test CXX_HAS_FFAST_MATH - Success
-- Looking for sincosf
-- Looking for sincosf - found
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.1")
-- Checking for module 'libconfig++'
--   Found libconfig++, version 1.5
-- Found lame includes: /usr/include/lame/lame.h
-- Found lame library: /usr/lib/aarch64-linux-gnu/libmp3lame.so
-- Checking for module 'shout'
--   Found shout, version 2.4.6
-- Looking for SHOUT_TLS_AUTO
-- Looking for SHOUT_TLS_AUTO - found
-- Looking for SHOUT_TLS_AUTO_NO_PLAIN
-- Looking for SHOUT_TLS_AUTO_NO_PLAIN - found
-- Looking for SHOUT_TLS_RFC2818
-- Looking for SHOUT_TLS_RFC2818 - found
-- Looking for SHOUT_TLS_RFC2817
-- Looking for SHOUT_TLS_RFC2817 - found
-- Looking for SHOUT_TLS_DISABLED
-- Looking for SHOUT_TLS_DISABLED - found
-- Looking for shout_set_tls
-- Looking for shout_set_tls - found
-- Checking for module 'librtlsdr'
--   Found librtlsdr, version 0.6.0
-- Found RTLSDR: /usr/lib/aarch64-linux-gnu/librtlsdr.so
-- Checking for module 'libmirisdr'
--   Package 'libmirisdr', required by 'virtual:world', not found
-- Could NOT find MiriSDR (missing: MIRISDR_LIBRARY MIRISDR_INCLUDE_DIR)
-- Checking for SoapySDR
-- Could NOT find SoapySDR (missing: SoapySDR_DIR)
--   SoapySDR not found
-- Checking for module 'libpulse'
--   Package 'libpulse', required by 'virtual:world', not found
-- Checking for module 'fftw3f'
--   Found fftw3f, version 3.3.10
-- RTLSDR-Airband configuration summary:

-- - Build type:                Release
-- - Operating system:          Linux
-- - SDR drivers:
--   - librtlsdr:               requested: ON, enabled: TRUE
--   - mirisdr:                 requested: ON, enabled: FALSE
--   - soapysdr:                requested: ON, enabled: FALSE
-- - Other options:
--   - Platform:                armv8-generic
--   - Build Unit Tests:        FALSE
--   - Broadcom VideoCore GPU:  FALSE
--   - NFM support:             OFF
--   - PulseAudio:              requested: ON, enabled: FALSE
--   - Profiling:               requested: OFF, enabled: FALSE
--   - Icecast TLS support:     TRUE
-- Configuring done
-- Generating done
-- Build files have been written to: /home/jwinship/RTLSDR-Airband-4.1.1/build

**Reply by win7463** [reply]
Same results. Will try the unstable branch now.

**Reply by win7463** [reply]
Compilied without errors following the instructions on the wiki for the latest development code.

jwinship@raspberrypi:~/RTLSDR-Airband/RTLSDR-Airband/build $ make
[  5%] Generating version.cpp, _version.cpp
[ 11%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/config.cpp.o
[ 16%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/input-common.cpp.o
[ 22%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/input-file.cpp.o
[ 27%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/input-helpers.cpp.o
[ 33%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/mixer.cpp.o
[ 38%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/output.cpp.o
[ 44%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/rtl_airband.cpp.o
[ 50%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/squelch.cpp.o
[ 55%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/ctcss.cpp.o
[ 61%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/util.cpp.o
[ 66%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/udp_stream.cpp.o
[ 72%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/logging.cpp.o
[ 77%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/filters.cpp.o
[ 83%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/helper_functions.cpp.o
[ 83%] Generating version.cpp, _version.cpp
[ 88%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/version.cpp.o
[ 94%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/input-rtlsdr.cpp.o
[ 94%] Built target rtl_airband_base
[100%] Linking CXX executable rtl_airband
[100%] Built target rtl_airband
jwinship@raspberrypi:~/RTLSDR-Airband/RTLSDR-Airband/build $ sudo make install
[  5%] Generating version.cpp, _version.cpp
[ 11%] Building CXX object src/CMakeFiles/rtl_airband_base.dir/version.cpp.o
[ 94%] Built target rtl_airband_base
[100%] Linking CXX executable rtl_airband
[100%] Built target rtl_airband
Install the project...
-- Install configuration: "Release"
-- Installing: /usr/local/bin/rtl_airband

**Reply by win7463** [reply]
This is the version reported. Hopefully it is the unstable build.


jwinship@raspberrypi:~/RTLSDR-Airband/RTLSDR-Airband/build $ rtl_airband -v
RTLSDR-Airband version 4.1.1-471abcf

**Reply by charlie-foxtrot** [reply]
Thanks for the info.  I'm guessing you're running `Raspberry Pi OS 12 (bookworm)`?

I'll get a release out ASAP as I'm betting more people will run into this shortly

**Reply by win7463** [reply]
Yes, running bookworm, the OS lite 64bit image.

**Comment by charlie-foxtrot** [comment]
`main` is now `v4.2.0` that includes #382
